Sources/SuperwallKit/Misc/Extensions/UIWindow+Swizzle.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  File.swift
+//  
+//
+//  Created by Yusuf Tör on 01/09/2023.
+//
+
+import UIKit
+
+extension UIWindow {
+  static var hasTouched = false
+  private static var swizzleSemaphore = DispatchSemaphore(value: 1)
+
+  /// Intercepts the very first touch on the UIWindow
+  static func swizzleSendEvent() {
+    let originalSelector = #selector(UIWindow.sendEvent(_:))
+    let swizzledSelector = #selector(swizzledSendEvent(_:))
+
+    guard
+        let originalMethod = class_getInstanceMethod(UIWindow.self, originalSelector),
+        let swizzledMethod = class_getInstanceMethod(UIWindow.self, swizzledSelector)
+    else {
+        return
+    }
+
+    method_exchangeImplementations(originalMethod, swizzledMethod)
+  }
+
+  @objc private func swizzledSendEvent(_ event: UIEvent) {
+    // TODO: This will block the main thread if multiple windows
+    Self.swizzleSemaphore.wait()
```

Comment:
Can you avoid the semaphore? I can't quit figure out what its purpose is

Reply:
This was more added as a test than what I actually think we should do. It should also be combined with a Bool but essentially if some apps have multiple `UIWindows`, this method could get called simultaneously and `touches_began` tracked twice. Need to avoid that but unsure how likely that scenario is.

Reply:
Looked at this again. `UIWindow` functions will always be performed on the `MainActor`, which means they'll be serialised by default, so we don't need to serialise it with a semaphore. Removed this now.